CONTRIBUTING.md

```diff
@@ -9,23 +9,32 @@ There are 4 main ways of contributing to the PyMC project (in descending order o
 * Contributing or improving the documentation (`docs`) or examples (`pymc/examples`)
 * Submitting issues related to bugs or desired enhancements
 
-# Opening issues
+## Opening issues
 
 We appreciate being notified of problems with the existing PyMC code. We prefer that issues be filed the on [Github Issue Tracker](https://github.com/pymc-devs/pymc/issues), rather than on social media or by direct email to the developers.
 
 Please verify that your issue is not being currently addressed by other issues or pull requests by using the GitHub search tool to look for key words in the project issue tracker.
 
 Filter on the ["beginner friendly"](https://github.com/pymc-devs/pymc/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22) label for issues which are good for new contributors.
 
-# Contributing code via pull requests
+## Etiquiette for code contributions
+When you start working working on an issue, open a `Draft` pull request as soon as you're doing the first steps.
+Make sure to mention the issue number in the PR description or commit message.
```

Comment:
```suggestion
Make sure to mention the issue number in the PR description and commit message.
```

Reply:
Why the duplication? I think I have never used issue numbers in commit messages, should I have been doing that?

Reply:
You can do that. It does make issue threads awfully verbose if a PR has a lot of rebases + force pushes, because a new automatic message is added every time that is done.

Reply:
If I always add the links to all relevant issues in the PR description, is there an advantage in adding them also in commit messages?

Reply:
Not really. You can always go from a commit to its PR and then to the original issue

Reply:
The links from the description can get lost depending on how the PR is merged.
> You can always go from a commit to its PR and then to the original issue

Yes, but not the other way around unless the issue number ends up in a commit message on `main`.
Or at least only with a detour through the PR.